src/app/child-dev-project/children/demo-data-generators/demo-child-generator.service.ts

```diff
@@ -31,36 +32,48 @@ export class DemoChildGenerator extends DemoDataGenerator<Entity> {
     ];
   }
 
-  static generateEntity(id: string) {
-    const child = createEntityOfType("Child", id);
-    child.name = faker.person.firstName() + " " + faker.person.lastName();
-    child.projectNumber = id;
-    child.religion = faker.helpers.arrayElement(religions);
-    child.gender = faker.helpers.arrayElement(genders.slice(0, 2));
-    child.dateOfBirth = new DateWithAge(
-      faker.date.birthdate({ mode: "age", min: 5, max: 20 }),
-    );
-    child.motherTongue = faker.helpers.arrayElement(languages);
-    child.center = faker.helpers.arrayElement(centersWithProbability);
-    child.phone =
-      "+" +
-      faker.number.int({ min: 10, max: 99 }) +
-      " " +
-      faker.number.int({ min: 10000000, max: 99999999 });
+  constructor(public config: DemoChildConfig) {
+    super();
+  }
 
-    child.admissionDate = faker.date.past({
-      years: calculateAge(child.dateOfBirth) - 4,
-    });
+  generateEntities(): Entity[] {
+    return generateChildren({ count: this.config.count });
+  }
+}
 
-    child["address"] = faker.geoAddress();
+function generateChildren(params: { count: number }): Entity[] {
+  return range(params.count).map((i) => generateChild(String(i + 1)));
+}
 
-    if (faker.number.int(100) > 90) {
-      DemoChildGenerator.makeChildDropout(child);
-    }
-    return child;
-  }
+/** @deprecated */
```

Comment:
```suggestion
/** @deprecated */
```
what should be used instead or why should it not be used? Let's add this to the "deprecated" comment

Reply:
Addressed in https://github.com/Aam-Digital/ndb-core/pull/3104/commits/c13f7e7274ee2b6ae83dc1b49461d11e0d58d84f